src/main/webapp/WEB-INF/tags/base.tag

```diff
@@ -31,12 +32,13 @@
     <a href="/conversations">Conversations</a>
     <a href="/activity">Activity Feed</a>
     <a href="/admin">Administration</a>
-    <% if(request.getSession().getAttribute("user") != null){ %>
-      <a>Hello <%= request.getSession().getAttribute("user") %>!</a>
-    <% } else{ %>
-      <a href="/login">Login</a>
+    <% if(request.getSession().getAttribute("user") == null){ %>
+          <a href="/login">Login</a>
     <% } %>
     <a href="/about.jsp">About</a>
+    <% if(request.getSession().getAttribute("user") != null){ %>
+      <a id="greeting">Hello <%= request.getSession().getAttribute("user") %>!</a>
+    <% } %>
```

Comment:
This allows us to have the greeting separately. Note the addition of the `id` attribute to the "Hello username" anchor

Reply:
This makes a lot of sense.